PWGCF/TwoParticleCorrelations/Tasks/r2p2-4-id.cxx

```diff
@@ -0,0 +1,444 @@
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Common/DataModel/PIDResponse.h"
+#include "Framework/ASoAHelpers.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+//-----PID Functions-----------------------------------------------------------
+template <typename T>
+bool NO_PID(T track)
+{
+  return true;
+}
+template <typename T>
+bool PID_PION(T track)
+{
+  float tpccut = 2.5, tofcut = 2.5;
+  if (fabs(track.tpcNSigmaPi()) >= tpccut)
+    return false;
+  if (track.pt() >= 0.6) {
+    if (track.hasTOF()) {
+      if (fabs(track.tofNSigmaPi()) >= tofcut)
+        return false;
+    } else {
+      return false;
+    }
+  }
+  return true;
+}
+template <typename T>
+bool PID_KAON(T track)
+{
+  float tpccut = 2, tofcut = 2;
+  if (track.pt() < 0.6) {
+    if (track.pt() < 0.45)
+      tpccut = 2;
+    else if (track.pt() < 0.55)
+      tpccut = 1;
+    else
+      tpccut = 0.6;
+    if (fabs(track.tpcNSigmaKa()) > tpccut)
+      return false;
+  } else if (track.hasTOF()) {
+    if ((fabs(track.tpcNSigmaKa()) > tpccut) || (fabs(track.tofNSigmaKa()) > tofcut))
+      return false;
+  } else {
+    return false;
+  }
+  return true;
+}
+template <typename T>
+bool PID_PROTON(T track)
+{
+  float tpccut = 2.2, tofcut = 2;
+  if (track.pt() < 1.1) {
+    if (track.pt() < 0.85)
+      tpccut = 2.2;
+    else
+      tpccut = 1;
+    if (fabs(track.tpcNSigmaPr()) > tpccut)
+      return false;
+  } else if (track.hasTOF()) {
+    if ((fabs(track.tpcNSigmaPr()) > tpccut) || (fabs(track.tofNSigmaPr()) > tofcut))
+      return false;
+  } else {
+    return false;
+  }
+  return true;
+}
+//-----------------------------------------------------------------------------
+template <typename T>
+bool (*pidarray[4])(T) = {NO_PID, PID_PION, PID_KAON, PID_PROTON}; // Array of PID functions
+
+namespace o2::aod
+{
+namespace idr2p2columns
+{
+DECLARE_SOA_COLUMN(BinFlag, binflag, bool); // To flag tracks without proper binning
+DECLARE_SOA_COLUMN(PID, pid, int);          // Flag for type of particle
+} // namespace idr2p2columns
+DECLARE_SOA_TABLE(Flags, "AOD", "Flags", idr2p2columns::BinFlag, idr2p2columns::PID);
+} // namespace o2::aod
+struct FillFlagsTable {
+  Produces<aod::Flags> ftable;
+  void process(soa::Join<aod::Tracks, aod::TrackSelection, aod::pidTPCPi, aod::pidTOFPi, aod::pidTPCPr, aod::pidTOFPr, aod::pidTPCKa, aod::pidTOFKa, aod::pidTPCEl, aod::pidTOFbeta, aod::TracksExtra> const& tracks)
+  {
+    bool binflag;
+    int etabin, phibin, pid;
+    for (auto track : tracks) {
+      etabin = (track.eta() + 0.8) * 15; // 15= 24/1.6
+      phibin = 36 * track.phi() / (2 * constants::math::PI);
+      if ((etabin < 0) || (etabin >= 24) || (phibin < 0) || (phibin >= 36))
+        binflag = false;
+      else
+        binflag = true;
+      pid = 0;
+      for (int i = 0; i < 4; i++) {
+        if (pidarray<decltype(track)>[i](track))
+          pid = pid * 10 + i;
+      }
```

Comment:
What about a track that passes both the pion and the kaon cuts?

Reply:
Then the PID column will store a value '12', indicating that the track has been identified as a both a Pion (1) and a Kaon (2).

Reply:
But that is basically wrong 
One track can only be of one species. It is wrong to identifying it as two or more
The multiplicity cannot be artificially increased considering one track as multiple tracks

Of course, a pion is a charged track and a pion. But for this case have a look at my previous comment in the track loop

Reply:
Yes, I understand. I've done the necessary changes now.